### PR TITLE
[typescript/en] Add doc for for-of and for-in loops in Typescript

### PR DESCRIPTION
**Pull request by ashraf-patel**
- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]`
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [x] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

### CODE REVIEW

**Comment by sauravexodus** [comment]
It might be a good idea to add an example for `for await` @ashraf-patel

File: typescript.html.markdown [hunk 1]
```diff
@@ -223,6 +223,30 @@ moreNumbers[5] = 5; // Error, elements are read-only
 moreNumbers.push(5); // Error, no push method (because it mutates array)
 moreNumbers.length = 3; // Error, length is read-only
 numbers = moreNumbers; // Error, mutating methods are missing
+
+// Iterators and Generators
+
+// for..of statement
+// iterate over the list of values on the object being iterated
+let arrayOfAnyType = [1, "string", false];
+for (let val of arrayOfAnyType) {
```
Comment (sauravexodus) [inline typescript.html.markdown]: If you are not mutating the variable then use `const`

File: typescript.html.markdown [hunk 2]
```diff
@@ -223,6 +223,30 @@ moreNumbers[5] = 5; // Error, elements are read-only
...
+    console.log(val); // 1, "string", false
+}
+
+let list = [4, 5, 6];
+for (let i of list) {
```
Comment (sauravexodus) [inline typescript.html.markdown]: Should use `const`

File: typescript.html.markdown [hunk 3]
```diff
@@ -223,6 +223,30 @@ moreNumbers[5] = 5; // Error, elements are read-only
...
+   console.log(i); // "4", "5", "6"
+}
+
+// for..in statement
+// iterate over the list of keys on the object being iterated
+for (let i in list) {
```
Comment (sauravexodus) [inline typescript.html.markdown]: You might want to use const here as well. Or include the value mutation example in the for block.

File: typescript.html.markdown (hunk 1 above)
Comment (ashraf-patel) [reply typescript.html.markdown]: Thanks, Done

**Comment by sauravexodus** [comment]
Looks good to me!

**Comment by divayprakash** [comment]
@ashraf-patel Thank you for the contribution and welcome to the LXIYM community!

**Comment by ashraf-patel** [comment]
@divayprakash Thank you :) Happy to contribute.

**Pull request merged**